include/zephyr/sys/uuid.h

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2024, SECO Mind Srl
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_SYS_UUID_H_
+#define ZEPHYR_INCLUDE_SYS_UUID_H_
+
+/**
+ * @file
+ *
+ * @brief Utility functions for the generation and parsing of Universal Unique Identifier.
+ * @details This driver is compliant with RFC9562: https://datatracker.ietf.org/doc/rfc9562/
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @defgroup uuid UUID
+ * @ingroup utilities
+ * @{
+ */
+
+/** @brief Number of bytes in the binary representation of a UUID. */
+#define UUID_SIZE 16
+
+/** @brief Length of the UUID canonical string representation, including the NULL terminator. */
+#define UUID_STR_LEN 37
+
+/** @brief Length of the UUID base64 string representation, including the NULL terminator. */
+#define UUID_BASE64_LEN 25
+
+/**
+ * @brief Length of the UUID base64 URL and filename safe string representation, including the
+ * NULL terminator.
+ */
+#define UUID_BASE64URL_LEN 23
+
+/** @brief Binary representation of a UUID. */
+typedef uint8_t uuid_t[UUID_SIZE];
+
+/**
+ * @brief Generate a UUIDv4.
+ *
+ * @param out The UUID where the result will be written.
+ */
+void uuid_generate_v4(uuid_t out);
+
+/**
+ * @brief Generate a UUIDv5.
+ *
+ * @details This function computes a deterministic UUID starting from a namespace UUID and binary
+ * data.
+ *
+ * @param[in] namespace The string representation of an UUID to be used as namespace.
+ * @param[in] data A pointer to the data that will be hashed to produce the UUID.
+ * @param[in] data_size The size of the data buffer.
+ * @param[out] out The UUID where the result will be written.
+ *
+ * @retval 0 if the UUID has been correctly generated and stored in @p out
+ * @retval -EINVAL if @p input is not acceptable
+ * @retval -ENOMEM if memory allocation failed
+ * @retval -ENOTSUP if mbedTLS returned an unrecognized error
+ */
+int uuid_generate_v5(const uuid_t namespace, const void *data, size_t data_size, uuid_t out);
+
+/**
+ * @brief Create a uuid_t from a binary formatted UUID.
+ *
+ * @param data The buffer where the binary UUID is stored in a big-endian order.
+ * @param data_size The size of the out buffer. Should be at least UUID_SIZE.
+ * @param out The UUID where the result will be written.
+ *
+ * @retval 0 if the UUID has been correctly parsed and stored in @p out
+ * @retval -EINVAL if @p data is null or @p data_size is too small
+ */
+int uuid_from_buffer(const uint8_t *data, size_t data_size, uuid_t out);
```

Comment:
Doesn't `data_size` always need to be 16 octets? 

We could consider
```suggestion
int uuid_from_buffer(const uint8_t data[UUID_SIZE], uuid_t out);
```